8.x/theme-development/client-side-cart-module.md

```diff
@@ -0,0 +1,52 @@
+## Client Side cart module
+
+[[toc]]
+
+The Larammerce Template Engine has the **local card module**. In this tutorial, the source code of this module is Reviewed.
+
+**local card module** is in the following directory `resources/assets/js/define/local_cart_service.js`.
+It has 437 lines of code and is based on require js but there is no requirement to use require js. To read more about it, refer to this address:[requirejs](https://www.requirejs.org/).
+
+Let's check the local card module:
+
+requirejs has two very important functions called define and require. In this module, there is a define function that takes three inputs:
+
+The first input is the name of the module, the second input is an arraylist of the dependencies of this module
+and the third input of the function body module.
+
+```js
+define('local_cart_service', ['jquery', 'jq_cookie', 'tools', 'template', 'underscore'],
+    function (jQuery, cookie, tools, template, _) {
+```
+
+In the next part, there are constants of the module.
+
+**cartCountEl** constant shows the number of products in the cart with the cart-count selector.
+```js
+ const cartCountEl = jQuery('.cart-count');
+```
+
+In the box where the product content is presented and the operations of adding to the shopping cart and removing from the shopping cart are performed, the **product-box** attribute must be given to its HTML tag.
+```js
+   const productSelector = '[product-box]';
+```
+
+This constant is the name of the cookie's key in which the shopping cart information is located.
+```js
+const cartCookie = window.siteEnv.SITE_LOCAL_CART_COOKIE_NAME;
+```
+
+The cookie has a specific capacity and more information cannot be stored in the cookie. In this constant, a limit is defined for the shopping cart.
+```js
+const cartCountLimit = window.siteEnv.SITE_LOCAL_CART_COUNT_LIMIT;
+```
+
+The discount amount applied to all card products.
```

Comment:
Here we need a more concise description:

As for every row of the cart, there is a calculation of discount in which each row calculates the amount by itself. Still, in case of any extra discount which must be applied to the whole invoice, there should be a way to store the extra discount amount somewhere out of the contents of the row. So there is a variable named `extraDiscountAmount` set externally by help of the method `LocalCartService.setExtraDiscount(amount)`.